build/bin/sage-spkg-info

```diff
@@ -111,16 +111,17 @@ if [ -z "$system" ]; then
 else
     echo
     if [ -f "$PKG_SCRIPTS"/spkg-configure.m4 ]; then
-        echo "If the system package is installed, ./configure will check whether it can be used."
-    else
-        echo "However, these system packages will not be used for building Sage"
         if [ -f "$PKG_SCRIPTS"/install-requires.txt ]; then
```

Comment:
This is not the right test because of packages like `tox`.

Reply:
Renaming the new kind of spkg-configure to `spkg-configure-sitepackage.m4` or something like that might simplify tests so that `grep` is no longer necessary to be invoked

Reply:
I thought about that or about adding some other piece of metadata like a new package type, but I think grep is the relatively "correct" option. The tangible difference between a "system site package" and any other package right now is the line,

```m4
AS_IF([test "${enable_system_site_packages}" = "yes"], [
```

in `sage_python_package_check.m4`. So in that sense, grepping for `SAGE_PYTHON_PACKAGE_CHECK` is the closest thing we have to a true indicator of whether or not some package needs `--enable-sytem-site-packages`.

Reply:
Sure, but the issue is that `sage -package list` now can no longer be used to distinguish between all the relevant package "types"

Reply:
I don't see any difference between a system python package and a system C (or whatever) package. The experimental option is there right now because it's going to turn on 100+ lightly tested system packages all at once, but the idea of using a python package from the system is not experimental. It used to be possible in sage, and is now a lot safer because we have `install-requires.txt` limiting their use to known-good versions. Once our doctests are updated, install-requires.txt are tightened, and there's a CI job for them, there will be no reason to treat them as a separate class of package.

Reply:
I disagree; I think `--enable-system-site-packages` is an option for a separate use case because it is incompatible with contemporary Python development practices -- until distributions find a proper way to support isolated venvs and build isolation.

Reply:
> until distributions find a proper way to support isolated venvs and build isolation

Yeah nobody wants that.

Reply:
> It used to be possible in sage

When? Not even system Python itself could be used before Sage 9.1

Reply:
> > until distributions find a proper way to support isolated venvs and build isolation
> 
> Yeah nobody wants that.

I know; but that's part of why distro Python is only relevant as a deployment target but not as a development environment.

Reply:
Anyway, this difference of opinion has no bearing on the present PR.

Reply:
It's the usual distinction between what's easy for developers and what's easy for users. Developers would love to completely ignore the issues of consistent versioning, API/ABI changes, security updates, disk/bandwidth waste, etc. And virtual environments and version pinning largely allow them to do that.

Users, on the other hand, want exactly the opposite. They want the versions to play nicely together, API/ABI upgrades to go smoothly, and all of their software to be patched and up to date without wasting any of their money. Since there are potentially thousands of users for every developer I'm of the opinion that the user should take preference. But regardless, all linux distributions exist solely to cater to their users. It's our job to help them achieve those goals. That's why I say none of us want to isolate anything. It's counterproductive for what we're trying to achieve.

> Anyway, this difference of opinion has no bearing on the present PR.

It kind of does. I see `--enable-system-site-packages` as a stopgap with the goal of eventually treating python packages like any other package. Converting hundreds of python packages to formal second-class citizens would make that a lot harder.

Reply:
> > It used to be possible in sage
> 
> When? Not even system Python itself could be used before Sage 9.1

Before the virtualenv. It wasn't something that happened automatically but you also didn't have to patch sage to make it possible.

Reply:
Well, we'll have to see how usable/maintainable  `--enable-system-site-packages` is going to be; which is why I support merging this PR. I don't insist on a cleaner implementation (that does not fake the package filtering in `sage-get-system-packages`) for this PR.

Reply:
Ok, thank you. I still have to take a shot at the `./configure` output when I can sit down and stare at it for a while. If the argument to `SAGE_SPKG_CONFIGURE` is always the name of the directory that `spkg-configure.m4` lives in, that may avoid the chicken-and-egg problem.